packages/server/src/middleware/migrations.ts

```diff
@@ -0,0 +1,143 @@
+import { context, locks, queue } from "@budibase/backend-core"
+import { UserCtx, Document, LockName, LockType } from "@budibase/types"
+import Bull, { Job } from "bull"
+
+export const MIGRATIONS: Record<string, { migration: () => Promise<void> }> = {
+  "20231122115100": {
+    migration: async () => {
+      await new Promise<void>(r => {
```

Comment:
we will need to work out the migration runner, as discussed yesterday this will likely need to involve a bit of: https://linear.app/budibase/issue/BUDI-7611/db-batching-functionality